### ISSUE TITLE
Add WriteJSI functionality for marshalling JSValues to JS

### ISSUE THREAD

**Issue by asklar** [microsoft/react-native-windows#7375]
Today when writing an event args to JS, one uses DispatchEvent; however this doesn't allow writing back JSI HostObjects, only allowing the simple types instead (int, string, bool, array, nested). 
The idea would be to allow a WriteJSI api that allows modules/apps to pass JSI HostObjects into the JSValueWriter

**Comment by NickGerleman** [comment]
We have this internally, currently used for NM2 TurboModules.

See https://github.com/microsoft/react-native-windows/blob/master/vnext/Microsoft.ReactNative/JsiWriter.h

**Comment by asklar** [comment]
yes, the problem is the DispatchEvent functionality relies on folly::dynamic today. Unclear if we can make this work without a core change.

**Comment by acoates-ms** [comment]
This may need to wait until we get rid of web debugging.
